src/blog/core2.clj

```diff
@@ -0,0 +1,63 @@
+(ns blog.core
+  (:require [clojure.java.io :as io]
+            [markdown.core :as md]
+            [clojure.string :as str]
+            [clojure.instant :as inst]))
+
+;; Set the directory for the blog post markdown files
+(def blog-dir "./blog/")
+
+;; Let's define the HTML that goes above the blog posts html
+(def html-header
+  (slurp (str blog-dir "static/header.html")))
+
+;; Let's define the HTML that goes below the blog posts html
+(def html-footer
+  (slurp (str blog-dir "static/footer.html")))
+
+; Let's attempt to load the directory for the posts directory
+#_(map #(.toString %) (.list (io/file posts-dir)))
+
+(defn get-md-files
+  "Return a sequence of all markdown files in dir"
+  [dir]
+  (->> (io/file dir)
+       (file-seq)
+       (map #(.toString %))
+       (filter #(str/ends-with? % ".md"))))
+
+
+;;(map slurp (get-md-files posts-dir))
+
+(defn process-blog-post
+  "Read the blog post markdown file and return a map where
+  the first line of the file is the :date keyword and the rest
+  of the file is the :content keyword"
+  [file]
+  (with-open [rdr (io/reader file)]
```

Comment:
Re: date, potential options to make it less hacky:
 - have the date in the filename `2023-10-21-foobar.md`
 - use markdown-clj support for [metadata](https://github.com/yogthos/markdown-clj#metadata)
 - use markdown "[frontmatter](https://markdoc.dev/docs/frontmatter)"

Reply:
(Then you'll be able to just slurp)